lldb/include/lldb/Symbol/CompilerType.h

```diff
@@ -324,7 +324,7 @@ class CompilerType {
 
   /// Lookup a child given a name. This function will match base class names and
   /// member member names in "clang_type" only, not descendants.
-  uint32_t GetIndexOfChildWithName(const char *name,
+  uint32_t GetIndexOfChildWithName(const char *name, ExecutionContext *exe_ctx,
```

Comment:
We should eventually upstream this change, but I don't see an issue with doing so. @jimingham previously pointed out that requiring a _ValueObject_ in CompilerType is problematic, but there is prior art for requiring an optional ExecutionContext in calls like GetBitSize(). @jimingham do you agree with that characterization?

Reply:
for what it's worth, the functions I've seen need the process, so passing a process instead of `ExecutionContext` makes the dependency clearer, at the cost of being less flexible if in the future something else from the context is needed instead of or in addition to the process.

Reply:
Since this is an API for all TypeSystems, maybe keeping it more general is actually desirable? It also looks like most CompilerType methods (https://lldb.llvm.org/cpp_reference/classlldb__private_1_1CompilerType.html) take an _ExecutionContextScope_. Maybe we should also do that here too, for symmetry?

Reply:
The use cases and distinction between `ExecutionContext` and `ExecutionContextScope` is not clear to me at all. I read their docs and looked at their APIs. The primary reason I went with `ExecutionContext` here is because from `ValueObject` it's more direct/convenient.

Within `ValueObject`, both of these patterns exist:


```cpp
ExecutionContext exe_ctx(GetExecutionContextRef());
```

```cpp
ExecutionContext exe_ctx(GetExecutionContextRef());
someFunction(exe_ctx.GetBestExecutionContextScope());
```

What's unclear to me, is why bring `ExecutionContextScope` into the equation? From its documentation, `ExecutionContext` says:

> These objects are designed to be used for short term execution context object storage while a function might be trying to evaluate something that requires a thread or frame.

Which does describe the needs of `SwiftLanguageRuntime`.

Reply:
> We should eventually upstream this change, but I don't see an issue with doing so. @jimingham previously pointed out that requiring a _ValueObject_ in CompilerType is problematic, but there is prior art for requiring an optional ExecutionContext in calls like GetBitSize(). @jimingham do you agree with that characterization?

That's right.  Since you really only want the process, or maybe a frame if you are trying to bind some generic, a ValueObject is way more than you need.  But that you might need a process to determine some aspects of the type system, it makes sense to have API's into the type system that provide an ExecutionContext in some form.

Reply:
> The use cases and distinction between ExecutionContext and ExecutionContextScope is not clear to me at all. I read their docs and looked at their APIs. 

Before I _guess_ an answer — @jimingham ^ ?

Reply:
As I understand it, the ExecutionContextScope is there because it is really handy to be able to use one of the objects in the Execution Context hierarchy to stand in for a scope specifier regardless of where you are in the hierarchy.  I'm not sure for local passing of contexts there's a hard and fast rule on which to use, so far as I can tell it's more a matter of convenience.  If you know you will mostly always have a process/thread/frame object on hand when calling the API, then ExecutionContextScope is way more convenient as you can bypass the conversion to ExecutionContext.  

But you might also want to check with Greg in case I'm misremembering something.  These were his designs originally.